sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,417 @@
+                    "type": str
+                    "is_unique": bool
+                    "is_primary_key": bool
+                    "is_unique_constraint": bool
+                    "is_disabled": bool,
+                    "column_names": str
+            foreign_keys : list of foreign key dicts
+                foreign_key
+                key/value:
+                    "foreign_key_name": str
+                    "referencing_table": str
+                    "referencing_column": str
+                    "referenced_table": str
+                    "referenced_column": str
+            partitions: partition dict
+                partition
+                key/value:
+                    "partition_count": int
+        """
+        if len(table_list) == 0:
+            return
+        name_to_id = {}
+        id_to_table_data = {}
+        table_ids_object = ",".join(["OBJECT_NAME({})".format(t.get("id")) for t in table_list])
+        table_ids = ",".join(["{}".format(t.get("id")) for t in table_list])
+        for t in table_list:
+            name_to_id[t["name"]] = t["id"]
+            id_to_table_data[t["id"]] = t
+        total_columns_number = self._populate_with_columns_data(
+            table_ids_object, name_to_id, id_to_table_data, schema, cursor
+        )
+        self._populate_with_partitions_data(table_ids, id_to_table_data, cursor)
+        self._populate_with_foreign_keys_data(table_ids, id_to_table_data, cursor)
+        self._populate_with_index_data(table_ids, id_to_table_data, cursor)
+        return total_columns_number, list(id_to_table_data.values())
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_columns_data(self, table_ids, name_to_id, id_to_table_data, schema, cursor):
+        cursor.execute(COLUMN_QUERY.format(table_ids, schema["name"]))
+        data = cursor.fetchall()
+        # AS default - cannot be used in sqlserver query as this word is reserved
+        columns = [
+            "default" if str(i[0]).lower() == "column_default" else str(i[0]).lower() for i in cursor.description
+        ]
+        rows = [dict(zip(columns, [str(item) for item in row])) for row in data]
+        for row in rows:
+            table_id = name_to_id.get(str(row.get("table_name")))
+            if table_id is not None:
+                row.pop("table_name", None)
+                if "nullable" in row:
+                    if row["nullable"].lower() == "no" or row["nullable"].lower() == "false":
+                        row["nullable"] = False
+                    else:
+                        row["nullable"] = True
+                if table_id in id_to_table_data:
+                    id_to_table_data.get(table_id)["columns"] = id_to_table_data.get(table_id).get("columns", []) + [
+                        row
+                    ]
+                else:
+                    self._log.error("Columns found for an unkown table with the object_id: %s", table_id)
```

Comment:
just a nit, but the log formatting in this file is inconsistent. Somethings we use `%s` and other times we use the `.format` function. Can you clean that up?